core/includes/form.inc

```diff
@@ -1389,8 +1389,8 @@ function _form_validate(&$elements, &$form_state, $form_id = NULL) {
           $value = in_array($elements['#type'], array('checkboxes', 'tableselect')) ? array_keys($elements['#value']) : $elements['#value'];
           foreach ($value as $v) {
             if (!isset($options[$v])) {
-              form_error($elements, $t('An illegal choice has been detected. Please contact the site administrator.'));
-              watchdog('form', 'Illegal choice %choice in !name element.', array('%choice' => $v, '!name' => empty($elements['#title']) ? $elements['#parents'][0] : $elements['#title']), WATCHDOG_ERROR);
+              form_error($elements, $t('Invalid option %choice in !name element.', array('%choice' => $v, '!name' => empty($elements['#title']) ? $elements['#parents'][0] : $elements['#title'])));
+              watchdog('form', 'Invalid option %choice in !name element.', array('%choice' => $v, '!name' => empty($elements['#title']) ? $elements['#parents'][0] : $elements['#title']), WATCHDOG_ERROR);
```

Comment:
I hadn't noticed that we use two different ways of displaying the form element title. Here we used `!name`, but below we used `%name` with the exact same string and input value. Let's switch to `%name` here, better to fully escape the form label or parent rather than assume it's safe.

Reply:
done.

Reply:
Thanks! Though the same test is still failing. We still need that fixed.

Reply:
yeah, I can't seem to figure out why it's failing.  Maybe I don't understand  the proper use of `format_string()`.